pkg/balancer/least_connections.go

```diff
@@ -0,0 +1,128 @@
+package balancer
+
+import (
+	"math"
+	"net"
+	"sync"
+	"time"
+
+	"github.com/square/p2/pkg/util"
+)
+
+type lcBackend struct {
+	conns    int
+	disabled bool
+}
+
+type LeastConnections struct {
```

Comment:
I know this is just an experiment, but any reason you didn't just round robin? If a host is somehow in the pool and connections fail immediately, this lb strategy would send all new requests to it (because it has the least connections)